editoast/src/views/train_schedule/projection.rs

```diff
@@ -277,40 +274,40 @@ async fn project_path(
     let signal_updates = signal_updates?;
 
     // 3. Store the projection in the cache (using pipeline)
-    let trains_hash_values: HashMap<_, _> = trains
+    let trains_hash_values: HashMap<_, _> = trains_details
         .iter()
-        .map(|t| t.id)
+        .map(|t| t.train_id)
         .zip(trains_hash_values)
         .collect();
     let mut new_items = vec![];
-    for id in miss_cache.keys() {
-        let hash = &trains_hash_values[id];
+    for id in miss_cache.iter().map(|t| t.train_id) {
```

Comment:
I'd argue that you might want to actually be explicit about which kind of identifier it is (I'm under the impression that it's actually because of that kind of things that the bug exists in the first place?).

```suggestion
    for train_id in miss_cache.iter().map(|t| t.train_id) {
```

Reply:
Fixed